### ISSUE TITLE
SecondaryNav passes all props to NavItem

### ISSUE THREAD

**Issue by calummoore** [mineral-ui/mineral-ui#827]
### Expected behavior
All props from `SecondryNav` should not be passed to `NavItem`.
```
<SecondryNav style={{ padding: 0 }} role='test'>
  <NavItem as={Link} to={url}>Demo<NavItem>
</SecondryNav> 
```

### Current behavior
All props are passed from `SecondayNav` to `NavItem`. As you can see from the screenshot, both `role` and `style` are passed to `NavItem`

<img width="1050" alt="Screenshot 2019-03-27 at 01 28 55" src="https://user-images.githubusercontent.com/4988117/55043808-c593e600-502f-11e9-91c4-31531945db37.png">

### Version
"mineral-ui": "^0.57.0",

### Environment
create-react-app

**Comment by kylegach** [comment]
Thanks for the bug report, @calummoore. I've submitted a PR (#828) with a fix. Feel free to review and leave comments there.

**Comment by kylegach** [comment]
@calummoore — This is fixed in [v0.58](https://www.npmjs.com/package/mineral-ui/v/0.58.0) (docs update for #826 has [also been published](https://mineral-ui.com/theming#typography)).
